src/colmap/estimators/cost_functions.h

```diff
@@ -114,35 +115,32 @@ class ReprojErrorConstantPoseCostFunction {
   bool operator()(const T* const point3D,
                   const T* const camera_params,
                   T* residuals) const {
-    const Eigen::Matrix<T, 3, 1> point3D_in_cam =
-        cam_from_world_.rotation.cast<T>() * EigenVector3Map<T>(point3D) +
+    const Eigen::Quaternion<T> cam_from_world_rotation =
+        cam_from_world_.rotation.cast<T>();
+    const Eigen::Matrix<T, 3, 1> cam_from_world_translation =
         cam_from_world_.translation.cast<T>();
-    CameraModel::ImgFromCam(camera_params,
-                            point3D_in_cam[0],
-                            point3D_in_cam[1],
-                            point3D_in_cam[2],
-                            &residuals[0],
-                            &residuals[1]);
-    residuals[0] -= T(observed_x_);
-    residuals[1] -= T(observed_y_);
-    return true;
+    return Parent::operator()(cam_from_world_rotation.coeffs().data(),
```

Comment:
It may be good to add an explicit "inline" in the parent or have you confirmed that there is no noticable perf impact with this change? I remember vaguely playing around with similar changes a long time ago but always ended up going back to the more repetitive code, as the compilers back then were not smart enough to properly inline and produce the same efficient code.